adapters/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/RestAdapterGenerator_ActionMapperErrorHandlingTest.java

```diff
@@ -0,0 +1,160 @@
+        exception.expect(RamlValidationException.class);
+        exception.expectMessage("Invalid action mapping in RAML file");
+
+        generator.run(
+                restRamlWithDefaults()
+                        .with(resource("/user")
+                                .with(defaultGetAction()
+                                        .withDescription("........ aaa incorrect mapping")
+                                )
+
+                        ).build(),
+                configurationWithBasePackage(BASE_PACKAGE, outputFolder, ACTION_MAPPING_TRUE));
+    }
+
+    @Test
+    public void shouldThrowExceptionIfMappingEmpty() throws Exception {
+        exception.expect(RamlValidationException.class);
+        exception.expectMessage("Invalid action mapping in RAML file");
+
+        generator.run(
+                restRamlWithDefaults()
+                        .with(resource("/user")
+                                .with(defaultGetAction()
+                                        .withDescription("...\n...\n")
+                                )
+
+                        ).build(),
+                configurationWithBasePackage(BASE_PACKAGE, outputFolder, ACTION_MAPPING_TRUE));
+    }
+
+    @Test
+    public void shouldThrowExceptionIfMappingNull() throws Exception {
+        exception.expect(RamlValidationException.class);
+        exception.expectMessage("Invalid action mapping in RAML file");
+
+        generator.run(
+                restRamlWithDefaults()
+                        .with(resource("/user")
+                                .with(httpAction()
+                                        .withHttpActionType(GET)
+                                        .withResponseTypes("application/vnd.ctx.query.defquery+json")
+                                )
+                        ).build(),
+                configurationWithBasePackage(BASE_PACKAGE, outputFolder, ACTION_MAPPING_TRUE));
+    }
+
+
+    @Test
+    public void shouldThrowExceptionIfNameNotInMapping() throws Exception {
+        exception.expect(RamlValidationException.class);
+        exception.expectMessage("Invalid RAML file. Action name not defined in mapping");
+
+        generator.run(
+                restRamlWithDefaults()
+                        .with(resource("/user")
+                                .with(defaultGetAction()
+                                        .withDescription("...\n" +
+                                                "(mapping):\n" +
+                                                "    responseType: application/vnd.structure.command.test-cmd+json\n" +
+                                                "...\n")
+                                )
+
+                        ).build(),
+                configurationWithBasePackage(BASE_PACKAGE, outputFolder, ACTION_MAPPING_TRUE));
+    }
+
+    @Test
+    public void shouldThrowExceptionIfNoMediaTypeSetInMapping() throws Exception {
```

Comment:
Do we have a tes for a generally valid mapping but with a typo somewhere, like an additional dot somewhere? does the code handle this elegantly?